pkg/querysharding/analysis.go

```diff
@@ -0,0 +1,137 @@
+package querysharding
+
+var excludedLabels = []string{"le"}
+
+type QueryAnalysis struct {
+	// Labels to shard on
+	shardingLabels []string
+
+	// When set to true, sharding is `by` shardingLabels,
+	// otherwise it is `without` shardingLabels.
+	shardBy bool
+}
+
+func nonShardableQuery() QueryAnalysis {
+	return QueryAnalysis{
+		shardingLabels: nil,
+	}
+}
+
+func newShardableByLabels(labels []string, by bool) QueryAnalysis {
+	labels = without(labels, excludedLabels)
+
+	return QueryAnalysis{
+		shardBy:        by,
+		shardingLabels: labels,
+	}
+}
+
+func (q QueryAnalysis) scopeToLabels(labels []string, by bool) QueryAnalysis {
```

Comment:
Can we use pointer receiver instead? This way we can avoid copying the object when returning the value.

Reply:
Will change to pointer receiver.